src/foam/core/reflow/Upload.js

```diff
@@ -420,6 +427,15 @@ foam.CLASS({
 
       var sink = this.bulkUpload ? {
         put: async function(o) {
+          // Apply object adaptation callback
+          if ( self.objectAdaption ) {
```

Comment:
You could just make the default: function() {} and then not bother with the if(). Why is it async?